deeplake/core/tests/test_query.py

```diff
@@ -49,7 +49,8 @@ def test_multi_source_query(hub_cloud_dev_token):
         'SELECT * FROM (SELECT * FROM "hub://activeloop/mnist-train" UNION (SELECT images, labels FROM "hub://activeloop/cifar100-train")) WHERE labels == 0',
         token=hub_cloud_dev_token,
     )
-    assert len(ds) == 6423
```

Comment:
I get why the result of this test was changed. Can you please add the following test too?
```
ds = deeplake.query(
    'SELECT * FROM "hub://activeloop/mnist-train" WHERE labels == 0 UNION (SELECT images, labels FROM "hub://activeloop/cifar100-train" WHERE labels == 0)',
        token=hub_cloud_dev_token,
    )
assert len(ds) == 6423
```